dart_sdk/BUILD.gn

```diff
@@ -0,0 +1,24 @@
+import("//flutter/common/config.gni")
```

Comment:
We want to allow Dart SDK hackers to test Dart SDK builds with local source changes, so we may end up with this dependency again. If it's better to have it split out into a standalone `BUILD.gn` I think I'd rather keep it.

Reply:
> We want to allow Dart SDK hackers to test Dart SDK builds with local source changes, so we may end up with this dependency again. If it's better to have it split out into a standalone `BUILD.gn` I think I'd rather keep it.

I don't follow how that's related. The existing way to hack on the Dart SDK in the engine tree is to make changes under `third_party/dart`, and to use the `--no-prebuilt-dart-sdk` flag to the `gn` script. Does that not work for some reason?

Reply:
If the `dart-sdk` target is in `flutter/BUILD.gn`, then the wasm build cannot consume or use it because the wasm build can't actually touch `flutter/BUILD.gn` without pulling in transitive dependencies that break under a wasm build. The transitive dependencies break even if you aren't building any of their targets, and they are dependencies we don't necessarily have control over.

That being said, we aren't actually using the dart-sdk target for the web build anymore, we just use prebuilts. I also don't actually see us adding a dependency on the `dart-sdk` target in the future either, and if we did it would be in a different manner that would probably build it with the non-default toolchain, since it's weird to pull host artifacts into the wasm build tree. So actually for our purposes I don't care anymore, and if you want me to refactor that target back into `flutter/BUILD.gn` I don't have any real practical objection, but I don't actually understand why that would be desirable. `flutter/BUILD.gn` is a bit of a kitchen sink and probably would be easier to work with if it were broken up into smaller logical components, and so having the `dart-sdk` separated into a different file seems better to me than keeping it in `flutter/BUILD.gn`. But again we don't really depend on either any more, so I don't mind either way you want to go with it.